rasa/utils/train_utils.py

```diff
@@ -335,3 +342,97 @@ def override_defaults(
                 config[key] = custom[key]
 
     return config
+
+
+def create_data_generators(
+    model_data: RasaModelData,
+    batch_sizes: Union[int, List[int]],
+    epochs: int,
+    batch_strategy: Text = SEQUENCE,
+    eval_num_examples: int = 0,
+    random_seed: Optional[int] = None,
+) -> Tuple[RasaBatchDataGenerator, Optional[RasaBatchDataGenerator]]:
+    """Create data generators for train and optional validation data.
+
+    Args:
+        model_data: The model data to use.
+        batch_sizes: The batch size(s).
+        epochs: The number of epochs to train.
+        batch_strategy: The batch strategy to use.
+        eval_num_examples: Number of examples to use for validation data.
+        random_seed: The random seed.
+
+    Returns:
+        The training data generator and optional validation data generator.
+    """
+    validation_data_generator = None
+    if eval_num_examples > 0:
+        model_data, evaluation_model_data = model_data.split(
+            eval_num_examples, random_seed,
+        )
+        validation_data_generator = RasaBatchDataGenerator(
+            evaluation_model_data,
+            batch_size=batch_sizes,
+            epochs=epochs,
+            batch_strategy=batch_strategy,
+            shuffle=True,
+        )
+
+    data_generator = RasaBatchDataGenerator(
+        model_data,
+        batch_size=batch_sizes,
+        epochs=epochs,
+        batch_strategy=batch_strategy,
+        shuffle=True,
+    )
+
+    return data_generator, validation_data_generator
+
+
+def create_common_callbacks(
+    epochs: int,
+    tensorboard_log_dir: Optional[Text] = None,
+    tensorboard_log_level: Optional[Text] = None,
+    checkpoint_dir: Optional[Path] = None,
+) -> List[tf.keras.callbacks.Callback]:
```

Comment:
As this method will be called for all our custom models, currently three, I just wanted to extract the code in a central place so that we don't repeat the logic three times.

Reply:
So you are basically saying to just remove the function here and copy over the logic to the model itself when it creates the callbacks?

Reply:
So far I see that it's called only once in `diet_classifier.py` but maybe I missed something. 
In this case, it might even be better to place this logic somewhere in the classifier itself (`DIETClassifier` or its parents if it makes sense) — sorry, I am sort of trying to avoid files like `*_utils.py` as much as I can especially when the logic is appicable to one particular component.

Feel free to skip it if my assumption is incorrect and if you see this particular code being in use in many different places, not just classifiers.

Reply:
it is also called in `TEDPolicy` so we cannot make it part of the diet classifier